SingularityUI/app/models/TaskResourceUsage.coffee

```diff
@@ -6,4 +6,18 @@ class TaskResourceUsage extends Model
 
     initialize: ({ @taskId }) =>
 
-module.exports = TaskResourceUsage
+    # Calculate CPU Usage by comparing previous usage to current usage
+    setCpuUsage: ->
+        previous = @get('previousUsage')
+        currentTime = @get('cpusSystemTimeSecs') + @get('cpusUserTimeSecs')
+        previousTime = previous.cpusSystemTimeSecs + previous.cpusUserTimeSecs
+        timestampDiff = @get('timestamp') - previous.timestamp
+        cpus_used = (currentTime - previousTime) / timestampDiff
+        cpuUsageExceeding = (cpus_used / @get('cpusLimit')) > 1.10
+        
+        if cpuUsageExceeding
```

Comment:
i think the additional states are a good idea, but let's implement them in a separate PR